### ISSUE TITLE
strategic planning: maps

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#2082]
ref: https://github.com/ArctosDB/arctos/issues/2011, https://github.com/ArctosDB/arctos/issues/2058, https://github.com/ArctosDB/arctos/issues/2081

Google Maps is tightly integrated with Arctos, and seemingly unrelated things (saving agents with certain kinds of addresses) rely heavily on behind-the-scenes API calls.

The GAPI works very well, is relatively easy to program, etc. I don't think there are any technical reasons to move away from it.

Google regularly deprecates products, requires code changes, etc. That can cause loss of functionality, or just a few days updating code every now and then.

Google has started requiring a credit card on file, and without some sort of waiver our usage might be a few hundred dollars per month. Calculating actual costs seems difficult to impossible, and navigating Google's business model is, at absolute best, frustrating. @mkoo is managing this for us, and we do currently have a cost waiver.

In short, the administrative side of Google is not as nice as the technical.

There are alternatives for some of our uses (https://leafletjs.com/ is popular for 'lightweight' maps, https://www.giscloud.com/ might do what Fusion+Maps does now, etc.). I know little about either the technical or administrative aspects of any of them, but in general smaller projects seem to be worse technically and better administratively.

Should we stay with Google or explore alternatives?

**Comment by dustymc** [comment]
Updating urgency. It seems that we don't have a waiver or it doesn't extend to the services we need, and our usage is thousands not hundreds.

See: https://github.com/ArctosDB/arctos/issues/2084

Elevation: https://elevation-api.io/

**Comment by dustymc** [comment]
https://www.bingmapsportal.com/ exists and has at least some of the functionality we need. Anyone have an existing agreement with MS that might allow access?

**Comment by mkoo** [comment]
Google Maps Platform is increasing our credit line--> in contact right now.
Dusty, care to estimate our usage needs?
https://developers.google.com/maps/premium/calculate-credits
https://credits-calculator.appspot.com/

**Comment by dustymc** [comment]
"estimate" is a very strong word for the data I have available, and future usage depends on things I don't know about yet (eg, incoming data) but...

<img width="1030" alt="Screen Shot 2019-05-20 at 10 00 42 AM" src="https://user-images.githubusercontent.com/5720791/58038942-3282a400-7ae6-11e9-963d-a2bc431121c1.png">

**Comment by amgunderson** [comment]
The "oops! something went wrong" is happening again. The SPNHC conference is happening next week with live demonstrations of Arctos, this needs to be fixed now, please!

**Comment by mkoo** [comment]
See Working Group email from last night. We are on it!

**Comment by mkoo** [comment]
We have just been given an increase in Maps Platform credit so we can
resume using the API keys!

**Comment by dustymc** [comment]
Yay!

Arctos is running with the keys you gave me; I think you need to give them permissions??

**Comment by dustymc** [comment]
Arctos maps and spatial services should all be running again.

@mkoo has secured what we hope to be sufficient credits to run the full array of Arctos spatial services.

I took a close look at leaflet (and various other things), and I am now certain that it would be a step backwards functionally. We can certainly do core map-things with open-source solutions, but it would be a lot of work to change over and there will be a loss of performance. 

Accessing things like elevation data are very similar - they're fairly low-quality, or they cost $$ (maybe both).

This seem like a good use case in moving towards becoming a legal entity.

**Comment by dustymc** [comment]
See also https://github.com/ArctosDB/arctos/issues/2092

An improvement to the current process would be to (somehow) spread the spread the load out, so we're not getting a yearly (or whatever we set the interval to) huge glump.
